store/mongo/datastore_mongo.go

```diff
@@ -1715,6 +1720,32 @@ func (db *DataStoreMongo) AbortDeviceDeployments(ctx context.Context,
 	return nil
 }
 
+func (db *DataStoreMongo) DeleteDeviceDeploymentsHistory(ctx context.Context,
+	deviceID string) error {
+	database := db.client.Database(mstore.DbFromContext(ctx, DatabaseName))
+	collDevs := database.Collection(CollectionDevices)
+	selector := bson.M{
+		StorageKeyDeviceDeploymentDeviceId: deviceID,
+		StorageKeyDeviceDeploymentActive:   false,
+		StorageKeyDeviceDeploymentDeleted: bson.M{
+			"$exists": false,
+		},
+	}
+
+	now := time.Now()
+	update := bson.M{
+		"$set": bson.M{
+			StorageKeyDeviceDeploymentDeleted: &now,
+		},
+	}
+
+	if _, err := collDevs.UpdateMany(ctx, selector, update); err != nil {
```

Comment:
doesn't really matter, but why `UpdateMany` insead of `UpdateOne`?

Reply:
Because this will (logically) delete multiple records, in particular all the device deployment records for a given device that are NOT active (historical data).

Reply:
right; sorry for the noise